test/request.go

```diff
@@ -84,6 +84,8 @@ func EventuallyMatchesBody(expected string) spoof.ResponseChecker {
 	}
 }
 
+type endpointStateCallable = func(ctx context.Context, u *url.URL, inState spoof.ResponseChecker, desc string, r ...spoof.RequestOption) (*spoof.Response, error)
```

Comment:
do we think there are ever likely to be more than two possible things "implementing" this? If not I'd be very tempted to just pass down a `wait bool` parameter here